src/tslint.ts

```diff
@@ -88,6 +91,21 @@ class Linter {
     private containsRule(rules: RuleFailure[], rule: RuleFailure) {
         return rules.some((r) => r.equals(rule));
     }
+
+    private computeFullOptions() {
+        const {configuration, configurationPath, rulesDirectory} = this.options;
+
+        if (configuration == null && configurationPath == null) {
+            throw new Error("An ILinterOptions object must have either `configuration` or `configurationPath` specified");
+        }
+
+        const configFromPath = configurationPath != null ? loadConfigurationFromPath(configurationPath) : {};
+        const configProvided = configuration != null ? configuration : {};
+        const finalConfig = extendConfigFile(configProvided, configFromPath);
+
+        this.options.configuration = finalConfig;
+        this.options.rulesDirectory = arrayify(rulesDirectory).concat(arrayify(finalConfig.rulesDirectory));
```

Comment:
This will overwrite rules directories supplied in this option, best to tweak this